azext_iot/operations/hub.py

```diff
@@ -557,95 +561,97 @@ def _iot_hub_configuration_create(cmd, config_id, content, hub_name=None, target
     if labels:
         labels = process_json_arg(labels, argument_name="labels")
 
-    if edge:
-        config_content = ConfigurationContent(modules_content=content)
-    else:
-        config_content = ConfigurationContent(device_content=content)
+    config_content = ConfigurationContent(**processed_content)
 
     config_metrics = ConfigurationMetrics(queries=metrics)
     config = Configuration(id=config_id,
-                           schema_version='2.0',
+                           schema_version="2.0",
                            labels=labels,
                            content=config_content,
                            metrics=config_metrics,
                            target_condition=target_condition,
-                           etag=None,
+                           etag="*",
                            priority=priority,
-                           content_type='assignment')
+                           content_type="assignment")
     try:
         return service_sdk.create_or_update_configuration(config_id, config)
     except errors.CloudError as e:
         raise CLIError(unpack_msrest_error(e))
 
 
-def _process_config_content(content, content_type='module'):
-    content_key = 'modulesContent' if 'module' in content_type else 'deviceContent'
-    legacy_key = 'moduleContent'
-
-    if 'content' in content:
-        content = content['content']
-
-    if content_key in content:
-        # Schema based validation currently for IoT edge deployment only
-        if content_key == 'modulesContent':
-            _filter_config_content(content)
-            _validate_payload_schema(content)
-
-        content = content[content_key]
-    elif 'module' in content_type and legacy_key in content:
-        logger.warning("'%s' is deprecated use '%s' instead - request is still processing...", legacy_key, content_key)
-        content = content[legacy_key]
-    else:
-        raise CLIError("content json must include the '{}' property".format(content_key))
-
-    return content
-
+def _process_config_content(content, config_type):
+    from knack.util import to_snake_case
+
+    # Supports scenario where configuration payload is contained in 'content' key
+    if "content" in content:
+        content = content["content"]
+
+    # Create new config dict to remove superflous properties
+    processed_content = {}
+    if config_type == ConfigType.adm:
+        valid_adm_keys = ["deviceContent", "moduleContent"]
```

Comment:
If a user submits both "deviceContent" and "moduleContent"we should error out rather than assuming they wanted deviceContent.